### PR TITLE
Oracle module getter functions

### PR DESCRIPTION
**Pull request by haythem96**
This PR add the getters functions in the Oracle contract.
This PR should be reviewed/merged after merging this pull request #53 .

### CODE REVIEW

File: contracts/Oracle.sol [hunk 1]
```diff
@@ -3,10 +3,68 @@
...
+        uint256 timestamp; // timestamp at which the price is pulled to this oracle
+    }
+
+    /// @dev mapping between oracle and it's locking period
+    mapping(address => uint256) internal oracleLockingPeriod;
+    /// @dev mapping of asset price to it's dispute period
+    mapping(address => uint256) internal oracleDisputePeriod;
+    /// @dev mapping between batch and it's oracle
+    mapping(bytes32 => address) internal batchOracle;
+    /// @dev mapping between batch and it price at specific timestmap
```
Comment (aparnakr) [inline contracts/Oracle.sol:28]: can you explain what a batch is here?

File: contracts/Oracle.sol [hunk 2]
```diff
@@ -3,10 +3,68 @@
...
+    mapping(bytes32 => mapping(uint256 => Price)) internal batchPriceAt;
+
+    /**
+     * @notice get batch price
+     * @param _batch batch hash
```
Comment (aparnakr) [inline contracts/Oracle.sol:33]: explain what a batch is

File: contracts/Oracle.sol [hunk 3]
```diff
@@ -3,10 +3,68 @@
...
+     * @param _timestamp price timestamp
+     * @return price and timestap at which price submitted to this contract
+     */
+    function getBatchPrice(bytes32 _batch, uint256 _timestamp) public view returns (uint256, uint256) {
+        Price memory batchPrice = batchPriceAt[_batch][_timestamp];
+        return (batchPrice.price, batchPrice.timestamp);
+    }
+
+    /**
+     * @notice get batch oracle
```
Comment (aparnakr) [inline contracts/Oracle.sol:43]:
```suggestion
     * @notice get the price oracle for a specific batch. A batch is ... TODO: 
```

File: contracts/Oracle.sol [hunk 4]
```diff
@@ -3,10 +3,68 @@
...
+     * @param _batch batch hash
+     * @return oracle address
+     */
+    function getBatchOracle(bytes32 _batch) public view returns (address) {
+        return batchOracle[_batch];
+    }
+
+    /**
+     * @notice get oracle locking period
```
Comment (aparnakr) [inline contracts/Oracle.sol:52]: what is the oracle locking period? Is it the time up to which an oracle is locked? I'm confused by this

File: contracts/Oracle.sol [hunk 5]
```diff
@@ -3,10 +3,68 @@
...
+     * @dev during an oracle locking period, price can not be submitted to this contract
+     * @param _oracle oracle address
+     * @return locking period
+     */
+    function getOracleLockingPeriod(address _oracle) public view returns (uint256) {
+        return oracleLockingPeriod[_oracle];
+    }
 
+    /**
+     * @notice get oracle dispute period
```
Comment (aparnakr) [inline contracts/Oracle.sol:62]: what is the oracle locking period? Is it after the oracle locking period? Better description needed

File: contracts/Oracle.sol (hunk 4 above)
Comment (haythem96) [reply contracts/Oracle.sol:52]: I explained in the @dev, should I move it to @notice ?

File: contracts/Oracle.sol (hunk 5 above)
Comment (haythem96) [reply contracts/Oracle.sol:62]: I explained in the @dev, should I move it to @notice ?

File: contracts/Oracle.sol (hunk 4 above)
Comment (aparnakr) [reply contracts/Oracle.sol:52]: maybe in notice explain: Each underlying-collateral-strike-expiry has its own oracle

File: contracts/Oracle.sol (hunk 5 above)
Comment (aparnakr) [reply contracts/Oracle.sol:62]: well like what is the locking period? When does it happen? How long is it for? What are the different time periods?

Comment (haythem96) [reply contracts/Oracle.sol:62]: cool, for how long and what are different time, there is no specific value, can be any value that the owner set.

Comment (antoncoding) [reply contracts/Oracle.sol:62]: Period implies a duration of time, but here we only return `uint256` so it's vague how would that turn into a period. Maybe add something like 
```
@dev time after expiry that the oracle will be locked. During the locking period, price can not be submitted to this contract
```
And for the return pararm:`@return: time in second after expiry`

File: contracts/Oracle.sol [hunk 6]
```diff
@@ -25,6 +25,46 @@ contract Oracle is Ownable {
...
+        return oracleLockingPeriod[_oracle];
+    }
+
+    /**
+     * @notice get oracle dispute period
+     * @dev during an oracle dispute period, the owner of this contract can dispute the submitted price and modify it
+     * @param _oracle oracle address
+     * @return dispute period
+     */
+    function getOracleDisputePeriod(address _oracle) public view returns (uint256) {
```
Comment (antoncoding) [inline contracts/Oracle.sol:67]: Similar to above, add something to explain what the `uint256` means 
```
@dev time after locking period that the owner of this contract can dispute the submitted price and modify it
```

File: contracts/Oracle.sol (hunk 5 above)
Comment (haythem96) [reply contracts/Oracle.sol:62]: cool will add a better description, BTW the return is not time in second after expiry, it is just a period of time, like 15min, 1 hour.. etc, not a specific time.

**Pull request merged**